mock/mock_test.go

```diff
@@ -92,6 +92,34 @@ func (i *TestExampleImplementation) TheExampleMethodFuncType(fn ExampleFuncType)
 	return args.Error(0)
 }
 
+// MockTestingT mocks a test struct
+type MockTestingT struct {
+	Mock
```

Comment:
I would rather avoid using `mock` to mock a dependency for its own test :)

Let's implement this with a simple mock instead to keep the tests for the package depending on the package itself working correctly :)